docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,219 @@
+```js
+import Client, {Config} from 'shopify-buy';
+
+const config = new Config({
+  domain: 'your-shop-name.myshopify.com',
+  storefrontAccessToken: 'your-storefront-access-token'
+});
+
+const client = new Client(config);
+```
+
+### Fetching Products and Collections
+
+The functions for fetching products and collections are mostly the same. Major differences are:
+
+1.  The v1 functions take in a `Storefront ID` for fetching a product or collection by ID. A `Storefront ID` can be found under
+    the `API details` section on product and collection pages in the Shopify admin. 
+    `API details` will only be displayed if you have a private app with Storefront API access. For more details see the 
+    [Getting Started Guide](https://help.shopify.com/api/storefront-api/getting-started#authentication) for the Storefront API.
+
+2.  The functions now take an additional optional `query` argument that specifies the fields to query on the resource.
+    ```js
+    const query = productNodeQuery(['title', ['images', imageConnectionQuery()]];
+
+    client.fetchProduct('Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUzMTMwMjc=', query).then((product) => {
+      console.log(product); // Product with only title and images
+    });
+    ```
+    Fields and queries can be nested like so:
+    ```js
+    // Product query with title and variants, where variants have price and title
+    const query = productNodeQuery(['title', ['variants', variantConnectionQuery(['price', 'title'])]]);
+    ```
+    If the `query` argument isn't supplied, a default list of arguments will be used.
+    See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md#Client.Queries) for more details.
+
+3.  `fetchCollection()` and `fetchAllCollections()` no longer return the products in the collections by default.
+    To fetch products on the collections, use `fetchCollectionWithProducts()` and `fetchAllCollectionsWithProducts()`,
+    or specify products in the query function parameter.
+    ```js
+    const collectionId = 'Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI1NzY5NzczMQ=='
+    
+    // Use the built-in function
+    client.fetchCollectionWithProducts(collectionId).then((collection) => {
+      console.log(collection); // Collection with all default fields and products with all default fields.
+    });
+
+    // or specify a custom query
+    const query = collectionNodeQuery(['title', ['products', productConnectionQuery(['title'])]]);
+
+    client.fetchCollection(collectionId, query).then((collection) => {
+      console.log(collection); // Collection with only title and products with only title
+    });
+    ```
+    Note that `fetchCollectionWithProducts()` and `fetchCollectionsWithProducts()` do not take a `query` argument.
+
+4.  `fetchQueryProducts()` and `fetchQueryCollections()` query different fields and take an optional `query` argument.
```

Comment:
It's kind of confusing because it takes a query object that is the shopifyql stuff and another query arg that's the same as the other query arguments. I'll include an example here to try and make that clearer.